pkg/utils/resources/resources.go

```diff
@@ -67,10 +67,7 @@ func Merge(resources ...v1.ResourceList) v1.ResourceList {
 // constructing a new one for each sum like Merge
 func MergeInto(dest v1.ResourceList, src v1.ResourceList) v1.ResourceList {
 	if dest == nil {
```

Comment:
I think it will be better if we panic here. Otherwise, the behavior of this MergeInto is inconsistent and confusing: sometimes it will allocate an object for you and you need to use the return value, and other times it will just use the destination used and directly modify it.

Reply:
I tend to agree with you. Seems odd that we do this but I think that we probably shouldn't change this logic in the context of this PR